### ISSUE TITLE
How are batch gradients computed on embedding layers?

### ISSUE THREAD

**Issue by instigatorofawe** [tensorflow/tensorflow#44394]
Consider the following model, which is more or less a 12-dimensional vector lookup table with 10 rows, initialized to all zeros.

```
model = keras.models.Sequential()
model.add(keras.layers.Embedding(input_dim=10, output_dim=12, embeddings_initializer=keras.initializers.zeros))
model.compile(optimizer=keras.optimizers.SGD(),loss=keras.losses.MeanSquaredError())
```

I simply want it to train to the following data:
```
x = numpy.append(numpy.zeros(10000), numpy.ones(10000))
y = numpy.append(numpy.random.multivariate_normal(numpy.zeros(12), numpy.diag(numpy.ones(12)), 10000),
                 numpy.random.multivariate_normal(numpy.ones(12)*2, numpy.diag(numpy.ones(12)), 10000), axis=0)
model.fit(x,y,epochs=1,batch_size=1)
```

When the batch size is 1, the model behaves predictably; using stochastic gradient descent, we train the weights of the embedding layer towards the means of the two conditional distribution - i.e. model(0) tends towards [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], and model(1) tends toward [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2].

However, for batch sizes larger than 1, the weights of the model tend toward the average value of y, without respect for the value of x - i.e. both model(0) and model(1) tend toward [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1].

Why does this happen? If we just want a lookup table, we can alternately implement it using a dense layer with no bias term; this simply requires that we one-hot encode the input.

```
model = keras.models.Sequential()
model.add(keras.layers.Dense(12, input_shape=(10,), use_bias=False, kernel_initializer=keras.initializers.zeros))
model.compile(optimizer=keras.optimizers.SGD(),loss=keras.losses.MeanSquaredError())

x = tensorflow.one_hot(numpy.append(numpy.zeros(10000), numpy.ones(10000)), 10)
y = numpy.append(numpy.random.multivariate_normal(numpy.zeros(12), numpy.diag(numpy.ones(12)), 10000),
                 numpy.random.multivariate_normal(numpy.ones(12)*2, numpy.diag(numpy.ones(12)), 10000), axis=0)
model.fit(x,y,epochs=16,batch_size=128)
```

This behaves as I would expect, even when using mini-batch for training. So why doesn't it work correctly using the embedding layer?

**Comment by nikitamaia** [comment]
Adding the `contributions welcome` label to this issue for further investigation by the community. If you are interested in working on this issue, please leave a comment and I will assign it to you. Thanks!

**Comment by soheil** [comment]
I'm interested.

**Comment by buihoanganhtuan** [comment]
The behavior is still reproducible on tensorflow 2.9.1. Is there any plan to fix it?

**Comment by ElysianStorm** [comment]
I am interested in resolving this bug, can anyone assist me with it? I am fairly new to open source contributions, but am a quick learner and passionate to contribute in public.

**Comment by sushreebarsa** [comment]
@instigatorofawe To compute the batch gradients for an embedding layer, we first need to compute the gradient of the loss function with respect to the output of the embedding layer. This can be done using the standard back-propagation algorithm.

In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here.
Thank you!
